<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LC_HyperbolaPropertiesEditingWidget</class>
 <widget class="QWidget" name="LC_HyperbolaPropertiesEditingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hyperbola Properties</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupParams">
     <property name="title">
      <string>Standard Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridParams">
      <item row="0" column="0">
       <widget class="QLabel">
        <property name="text"><string>Center X:</string></property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="leCenterX"/>
      </item>
      <item row="0" column="2">
       <widget class="QLabel">
        <property name="text"><string>Y:</string></property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="leCenterY"/>
      </item>

      <item row="1" column="0">
       <widget class="QLabel">
        <property name="text"><string>Major Radius (a):</string></property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="leMajorRadius"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel">
        <property name="text"><string>Minor Radius (b):</string></property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="leMinorRadius"/>
      </item>

      <item row="2" column="0">
       <widget class="QLabel">
        <property name="text"><string>Ratio (b/a):</string></property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="leRatio"/>
      </item>

      <item row="3" column="0">
       <widget class="QLabel">
        <property name="text"><string>Rotation (deg):</string></property>
       </widget>
      </item>
      <item row="3" column="1" colspan="3">
       <widget class="QLineEdit" name="leAngle"/>
      </item>

      <item row="4" column="0">
       <widget class="QLabel">
        <property name="text"><string>Vertex X:</string></property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="leVertexX">
        <property name="readOnly"><bool>true</bool></property>
       </widget>
      </item>
      <item row="4" column="2">
       <widget class="QLabel">
        <property name="text"><string>Y:</string></property>
       </widget>
      </item>
      <item row="4" column="3">
       <widget class="QLineEdit" name="leVertexY">
        <property name="readOnly"><bool>true</bool></property>
       </widget>
      </item>

      <item row="5" column="0">
       <widget class="QLabel">
        <property name="text"><string>Start φ:</string></property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="leAngle1"/>
      </item>
      <item row="5" column="2">
       <widget class="QLabel">
        <property name="text"><string>End φ:</string></property>
       </widget>
      </item>
      <item row="5" column="3">
       <widget class="QLineEdit" name="leAngle2"/>
      </item>
      <item row="5" column="4">
       <widget class="QLabel">
        <property name="text"><string>(0,0 = full branch)</string></property>
       </widget>
      </item>

      <item row="6" column="0" colspan="4">
       <widget class="QCheckBox" name="cbReversed">
        <property name="text"><string>Left Branch</string></property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <widget class="QGroupBox" name="groupFoci">
     <property name="title">
      <string>Foci &amp; Point on Curve (Alternative Editing)</string>
     </property>
     <layout class="QGridLayout" name="gridFoci">
      <item row="0" column="0">
       <widget class="QLabel">
        <property name="text"><string>Focus 1 X:</string></property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="leFocus1X"/>
      </item>
      <item row="0" column="2">
       <widget class="QLabel">
        <property name="text"><string>Y:</string></property>
       </widget>
      </item>
      <item row="0" column="3">
       <widget class="QLineEdit" name="leFocus1Y"/>
      </item>

      <item row="1" column="0">
       <widget class="QLabel">
        <property name="text"><string>Focus 2 X:</string></property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="leFocus2X"/>
      </item>
      <item row="1" column="2">
       <widget class="QLabel">
        <property name="text"><string>Y:</string></property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="leFocus2Y"/>
      </item>

      <item row="2" column="0">
       <widget class="QLabel">
        <property name="text"><string>Point on Curve X:</string></property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="leStartX"/>
      </item>
      <item row="2" column="2">
       <widget class="QLabel">
        <property name="text"><string>Y:</string></property>
       </widget>
      </item>
      <item row="2" column="3">
       <widget class="QLineEdit" name="leStartY"/>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <widget class="QGroupBox" name="groupDerived">
     <property name="title">
      <string>Derived Properties (Read-only)</string>
     </property>
     <layout class="QGridLayout" name="gridDerived">
      <item row="0" column="0">
       <widget class="QLabel">
        <property name="text"><string>Eccentricity:</string></property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="leEccentricity">
        <property name="readOnly"><bool>true</bool></property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>

   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>